src/midgard/pointll.cc

```diff
@@ -290,6 +290,10 @@ PrecisionT GeoPoint<PrecisionT>::HeadingAtEndOfPolyline(const std::vector<GeoPoi
                     pt1->lat() + ((pt0->lat() - pt1->lat()) * pct));
         return ll.Heading(pts[idx1]);
       } else {
+        if (pt0 == pts.begin()) {
```

Comment:
ok so this happens when we never have enough total length to reach the distance threshold. can you make one small clean up here and remove the else case completely? it just adds more nesting where its not needed

Reply:
Indeed. Done.